<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Read QTL data — read.cross • MetaPipe</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Read QTL data — read.cross" />
<meta property="og:description" content="Read QTL data from two sources, one containing genotypes and another one
phenotypes." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MetaPipe</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.900</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/assess-normality.html">Assess Normality</a>
    </li>
    <li>
      <a href="../articles/load-raw-data.html">Load raw data</a>
    </li>
    <li>
      <a href="../articles/replace-missing-data.html">Replace Missing Data</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/villegar/MetaPipe/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Read QTL data</h1>
    <small class="dont-index">Source: <a href='https://github.com/villegar/MetaPipe/blob/master/R/utils-qtl.R'><code>R/utils-qtl.R</code></a></small>
    <div class="hidden name"><code>read.cross.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Read QTL data from two sources, one containing genotypes and another one
phenotypes.</p>
    </div>

    <pre class="usage"><span class='fu'>read.cross</span><span class='op'>(</span><span class='va'>geno</span>, <span class='va'>pheno</span>, wdir <span class='op'>=</span> <span class='fu'>here</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/here/man/here.html'>here</a></span><span class='op'>(</span><span class='op'>)</span>, quiet <span class='op'>=</span> <span class='cn'>TRUE</span>, <span class='va'>...</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>geno</th>
      <td><p>Data frame or string (filename) to file containing genotype 
data. For example a genetic map like <code><a href='father_riparia.html'>father_riparia</a></code>.</p></td>
    </tr>
    <tr>
      <th>pheno</th>
      <td><p>Data frame or string (filename) to file containing phenotype 
data. For example, output from <code><a href='assess_normality.html'>assess_normality</a></code>.</p></td>
    </tr>
    <tr>
      <th>wdir</th>
      <td><p>Working directory.</p></td>
    </tr>
    <tr>
      <th>quiet</th>
      <td><p>Boolean flag to hide status messages.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Arguments passed on to <code><a href='https://rdrr.io/pkg/qtl/man/read.cross.html'>qtl::read.cross</a></code></p><dl>
    <dt><code>file</code></dt><dd><p>The main input file for formats <code>csv</code>, <code>csvr</code>
    and <code>mm</code>.</p></dd>
    <dt><code>genfile</code></dt><dd><p>File with genotype data (formats <code>csvs</code>,
    <code>csvsr</code>, <code>karl</code>, <code>gary</code> and <code>mapqtl</code> only).</p></dd>
    <dt><code>mapfile</code></dt><dd><p>File with marker position information (all
    except the <code>csv</code> formats).</p></dd>
    <dt><code>phefile</code></dt><dd><p>File with phenotype data (formats <code>csvs</code>,
    <code>csvsr</code>, <code>karl</code>, <code>gary</code> and <code>mapqtl</code> only).</p></dd>
    <dt><code>chridfile</code></dt><dd><p>File with chromosome ID for each marker (<code>gary</code>
    format only).</p></dd>
    <dt><code>mnamesfile</code></dt><dd><p>File with marker names (<code>gary</code> format only).</p></dd>
    <dt><code>pnamesfile</code></dt><dd><p>File with phenotype names (<code>gary</code> format
    only).</p></dd>
    <dt><code>na.strings</code></dt><dd><p>A vector of strings which are to be interpreted as
    missing values (<code>csv</code> and <code>gary</code> formats only).  For the
    <code>csv</code> formats, these are interpreted globally
    for the entire
    file, so missing value codes in phenotypes must not be valid
    genotypes, and vice versa. For the <code>gary</code> format, these are
    used only for the phenotype data.</p></dd>
    <dt><code>genotypes</code></dt><dd><p>A vector of character strings specifying the genotype
    codes (<code>csv</code> formats only).  Generally this is a vector of
    length 5, with the elements corresponding to AA, AB, BB, not BB
    (i.e., AA or AB), and not AA (i.e., AB or BB).  <b>Note</b>: Pay
    careful attention to the third and fourth of these; the order of
    these can be confusing.    If you are trying to read 4-way cross data, your file must have
    genotypes coded as described below, and you need to set
    <code>genotypes=NULL</code> so that no re-coding gets done.</p></dd>
    <dt><code>alleles</code></dt><dd><p>A vector of two one-letter character strings (or four,
    for the four-way cross), to be used as labels for the two alleles.</p></dd>
    <dt><code>estimate.map</code></dt><dd><p>For all formats but <code>qtlcart</code>, <code>mapqtl</code>,
    and <code>karl</code>: if TRUE and marker positions are not included in
    the input files, the genetic map is estimated using the function
    <code><a href='https://rdrr.io/pkg/qtl/man/est.map.html'>est.map</a></code>.</p></dd>
    <dt><code>convertXdata</code></dt><dd><p>If TRUE, any X chromosome genotype data is
    converted to the internal standard, using columns <code>sex</code> and
    <code>pgm</code> in the phenotype data if they available or by inference
    if they are not.  If FALSE, the X chromsome data is read as is.</p></dd>
    <dt><code>error.prob</code></dt><dd><p>In the case that the marker map must be estimated:
   Assumed genotyping error rate used in the calculation of the
   penetrance Pr(observed genotype | true genotype).</p></dd>
    <dt><code>map.function</code></dt><dd><p>In the case that the marker map must be estimated:
   Indicates whether to use the Haldane, Kosambi,
   Carter-Falconer, or Morgan map function when converting genetic
   distances into recombination fractions. (Ignored if m &gt; 0.)</p></dd>
    <dt><code>BC.gen</code></dt><dd><p>Used only for cross type <code>"bcsft"</code>.</p></dd>
    <dt><code>F.gen</code></dt><dd><p>Used only for cross type <code>"bcsft"</code>.</p></dd>
    <dt><code>crosstype</code></dt><dd><p>Optional character string to force a particular cross type.</p></dd>
  
</dl></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Object of <code>cross</code> class for QTL mapping.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p>Other QTL mapping functions: 
<code><a href='qtl_perm_test.html'>qtl_perm_test</a>()</code>,
<code><a href='qtl_scone.html'>qtl_scone</a>()</code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>father_riparia</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>ionomics</span><span class='op'>)</span>
<span class='va'>ionomics_rev</span> <span class='op'>&lt;-</span> <span class='fu'>MetaPipe</span><span class='fu'>::</span><span class='fu'><a href='replace_missing.html'>replace_missing</a></span><span class='op'>(</span><span class='va'>ionomics</span>, 
                                          excluded_columns <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>2</span><span class='op'>)</span>,
                                          replace_na <span class='op'>=</span>  <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='va'>ionomics_normalised</span> <span class='op'>&lt;-</span> 
  <span class='fu'>MetaPipe</span><span class='fu'>::</span><span class='fu'><a href='assess_normality.html'>assess_normality</a></span><span class='op'>(</span><span class='va'>ionomics_rev</span>,
                             excluded_columns <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>2</span><span class='op'>)</span>,
                             out_prefix <span class='op'>=</span> <span class='st'>"ionomics"</span>,
                             transf_vals <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>2</span>, <span class='fu'><a href='https://rdrr.io/r/base/Log.html'>exp</a></span><span class='op'>(</span><span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>,
                             show_stats <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
<span class='va'>x_data</span> <span class='op'>&lt;-</span> <span class='fu'>MetaPipe</span><span class='fu'>::</span><span class='fu'>read.cross</span><span class='op'>(</span><span class='va'>father_riparia</span>, 
                               <span class='va'>ionomics_normalised</span><span class='op'>$</span><span class='va'>norm</span>,
                               genotypes <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"nn"</span>, <span class='st'>"np"</span>, <span class='st'>"--"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt;  --Read the following data:
#&gt; 	 166  individuals
#&gt; 	 1115  markers
#&gt; 	 7  phenotypes</div><div class='output co'>#&gt; <span class='warning'>Warning: Some markers at the same position on chr 1,4,5,7,8,9,10,12,14,15,16,17; use jittermap().</span></div><div class='output co'>#&gt;  --Cross type: f2 </div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Roberto Villegas-Diaz, Dilmini Alahakoon, Anne Fennell.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


